src/main/java/com/encora/BinarySearchTree.java

```diff
@@ -96,65 +81,85 @@ private Node delete(Node node, int value) {
         return node;
     }
 
-    private int smallestRightNumber(Node root) {
+    private T smallestRightNumber(Node<T> root) {
         /* Without recursion
         Node aux = root;
         while (aux.getLeft() != null) {
             aux = root.getLeft();
         }
         return aux.getValue();
          */
-        Node left = root.getLeft();
-        return left == null ? root.getValue() : smallestRightNumber(left);
+        Node<T> left = root.getLeft();
+        return left == null ? root.getData() : smallestRightNumber(left);
     }
 
-    public List<Integer> inOrder() {
+
+    private List<T> inOrder() {
         return inOrder(root);
     }
 
-    private List<Integer> inOrder(Node node) {
-        List<Integer> tree = new LinkedList<Integer>();
+    private List<T> inOrder(Node<T> node) {
+        List<@NonNull T> tree = new LinkedList<>();
         if (node != null) {
             tree.addAll(inOrder(node.getLeft()));
-            tree.add(node.getValue());
+            tree.add(node.getData());
             tree.addAll(inOrder(node.getRight()));
         }
         return tree;
     }
 
-    public List<Integer> preOrder() {
+    private List<T> preOrder() {
         return preOrder(root);
     }
 
-    private List<Integer> preOrder(Node node) {
-        List<Integer> tree = new LinkedList<Integer>();
+    private List<T> preOrder(Node<T> node) {
+        List<@NonNull T> tree = new LinkedList<>();
```

Comment:
Why do you need `@NonNull` here?
Ideally we should be good with validation at line 131/117.